test/index.test.ts

```diff
@@ -106,47 +106,48 @@ describe("mkdist", () => {
     expect(writtenFiles.sort()).toEqual(
       [
         "dist/README.md",
-        "dist/bar.d.ts",
+        "dist/bar.d.mts",
         "dist/bar.mjs",
         "dist/demo.css",
-        "dist/dir-export.d.ts",
+        "dist/dir-export.d.mts",
         "dist/dir-export.mjs",
         "dist/foo.mjs",
         "dist/foo.d.ts",
+        "dist/foo.d.mts",
         "dist/index.mjs",
-        "dist/index.d.ts",
+        "dist/index.d.mts",
         "dist/star/index.mjs",
-        "dist/star/index.d.ts",
+        "dist/star/index.d.mts",
         "dist/star/other.mjs",
-        "dist/star/other.d.ts",
+        "dist/star/other.d.mts",
         "dist/types.d.ts",
         "dist/components/index.mjs",
-        "dist/components/index.d.ts",
+        "dist/components/index.d.mts",
         "dist/components/blank.vue",
-        "dist/components/blank.vue.d.ts",
+        "dist/components/blank.vue.d.mts",
         "dist/components/js.vue",
-        "dist/components/js.vue.d.ts",
+        "dist/components/js.vue.d.mts",
         "dist/components/script-multi-block.vue",
-        "dist/components/script-multi-block.vue.d.ts",
+        "dist/components/script-multi-block.vue.d.mts",
         "dist/components/script-setup-ts.vue",
-        "dist/components/script-setup-ts.vue.d.ts",
+        "dist/components/script-setup-ts.vue.d.mts",
         "dist/components/ts.vue",
-        "dist/components/ts.vue.d.ts",
+        "dist/components/ts.vue.d.mts",
         "dist/components/jsx.mjs",
         "dist/components/tsx.mjs",
-        "dist/components/jsx.d.ts",
-        "dist/components/tsx.d.ts",
+        "dist/components/jsx.d.mts",
+        "dist/components/tsx.d.mts",
         "dist/bar/index.mjs",
-        "dist/bar/index.d.ts",
+        "dist/bar/index.d.mts",
         "dist/bar/esm.mjs",
         "dist/bar/esm.d.mts",
         "dist/ts/test1.mjs",
         "dist/ts/test2.mjs",
         "dist/ts/test1.d.mts",
-        "dist/ts/test2.d.cts",
+        "dist/ts/test2.d.mts",
```

Comment:
In this case we have a `test2.cts` in the `src`. I think there are two ways to handle it:

1. When a `.cts` or `.mts` file is found in the source, generate `.cjs/.d.cts` or `.mjs/.d.mts` pairs regardless of the `ext` option. This is the issue described by #138. Another variation would be to only override `ext` if it were not explicitly set.
2. Always use the `ext` option and treat `.mts`, `.cts`, and `.ts` files in `src/` identically.

 It's just important that the declaration and js file match, so you don't end up with `foo.cjs` and `foo.d.mts`